spec/core/file/setuid_spec.rb

```diff
@@ -28,15 +28,15 @@
     it "returns true when the gid bit is set" do
       platform_is :solaris do
         # Solaris requires execute bit before setting suid
-        NATFIXME 'Implement Kernel#system', exception: NoMethodError, message: "undefined method `system'" do
-          system "chmod u+x #{@name}"
-        end
-        `chmod u+x #{@name}`
+        system "chmod u+x #{@name}"
+        # NATFIXME: Fix arguments for Kernel#system
+        system "chmod", "u+x", @name
       end
-      NATFIXME 'Implement Kernel#system', exception: NoMethodError, message: "undefined method `system'" do
+      NATFIXME 'Fix arguments for Kernel#system', exception: SpecFailedException do
         system "chmod u+s #{@name}"
+        File.setuid?(@name).should == true
       end
-      `chmod u+s #{@name}`
+      system "chmod", "u+s", @name
```

Comment:
This is a bit of a mess now, since we can run the `system` method, but it doesn't do what it is supposed to do. I added a second check for `File.setuid?` into this block, so at least we get a notification once this starts working.

I'll try to look into the arguments of `Kernel.spawnf` soon-ish, before this spreads.